### PR TITLE
Hetzner failover IP: refactoring

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This is a follow-up to #56103 and taken out of #56186 to work around ansibot problems. It contains all refactoring steps, but does not yet add the new `hetzner_failover_ip_info` module. It creates a module_utils `hetzner` module, adds a module documentation fragment, and splits up the `get_failover` function into two.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
hetzner_failover_ip

### CODE REVIEW

File: lib/ansible/plugins/doc_fragments/hetzner.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class ModuleDocFragment(object):
+
+    # Standard files documentation fragment
+    DOCUMENTATION = r'''
+options:
+  hetzner_user:
+    description: The username for the Robot webservice user.
+    type: str
+    required: yes
+  hetzner_pass:
```
Comment (resmo) [inline lib/ansible/plugins/doc_fragments/hetzner.py]: hmm... I would prefer `hetzner_password`

**Comment by felixfontein** [comment]
CC @resmo @LKaemmerling

File: lib/ansible/plugins/doc_fragments/hetzner.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/plugins/doc_fragments/hetzner.py]: I've renamed it, and moved the common argspec entries to module_utils.

**Review by resmo — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
@resmo thanks for reviewing and merging!

**Pull request merged**